### ISSUE TITLE
Rebol binaries fail on NixOS

### ISSUE THREAD

**Issue by Ashvith10** [orx/orx#86]
While running `./setup.sh`, the shell script fails on NixOS:

```console
$ ./setup.sh
bash: ./code/build/rebol/r3-linux64: cannot execute: required file not found
```

Here's what I get when I run the `ldd` command:
```console
$ ldd code/build/rebol/r3-linux64
	linux-vdso.so.1 (0x00007ffdabfa6000)
	libdl.so.2 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libdl.so.2 (0x00007fd014ab7000)
	libm.so.6 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libm.so.6 (0x00007fd0149d7000)
	libc.so.6 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libc.so.6 (0x00007fd0147ef000)
	/lib64/ld-linux-x86-64.so.2 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib64/ld-linux-x86-64.so.2 (0x00007fd014abe000)
```

**Comment by enobayram** [comment]
Seems similar to [this issue](https://discourse.nixos.org/t/why-does-this-binary-executable-give-cannot-execute-required-file-not-found/33948), so I wonder whether you could work around it by running the setup inside `steam-run`.

**Comment by Ashvith10** [comment]
@enobayram I'd suspect it is more of a pathELF error, but I did not add it in the above post as I was unsure, since all the files seemed to be properly linked, at least from my perspective. The packaging method would be a nice fix, but the issue is both Rebol and Orx would have to be packaged. And the problem with the Rebol repository is that it is missing tags, which can be quite inconvenient, at least while trying to package apps on NixOS.

**Comment by iarwain** [comment]
@Ashvith10 Sorry I'm not very knowledgeable when it comes to Linux runtime issues.
Is there anything we can do at orx's level or should we relay this issue to Oldes/Rebol3?

**Comment by Ashvith10** [comment]
@iarwain apologies, I thought that you were using rebol packages from [rebol/rebol](https://github.com/rebol/rebol/). Looks like @Oldes has added tags for us, which is exactly what I was looking for. I'll create a simple rebol Nix expression and see how we can continue from there.

**Comment by Oldes** [comment]
Hi.. I've never heard about NixOS so I am not sure, what exactly do you need.. but if you want to compile Rebol (my fork) for a new system, you can use this [bootstrap](https://github.com/Siskin-framework/Rebol/tree/bootstrap) source.

Rebol sources are partially generated using Rebol itself.. the mentioned boostrap branch contains these generated files so one can just edit the makefile and make the initial version for the new OS.. and use this bootstrap version to make a real version.

**Comment by iarwain** [comment]
This probably means that the variable `platform`, set at the beginning of the script, isn't containing one of the expected values (`windows, `linux` or `mac`). I'll modify the script to make it more robust and fail more graciously.

However, in order to properly support NixOS, could you run the rebol interpreter and evaluate the following expression: `system/build/os`? What return value do you obtain?

**Comment by Oldes** [comment]
> @iarwain valid comment, although not related to this issue. Looks like the build thinks that I'm on Debian for some reason:
> 
> ```
> >> system/build/os
> == debian
> ```
> 
> Perhaps @Oldes can help me here with this. About the package thingy, is Rebol used anywhere else apart from the setup and init phase? I guess it is used for game data stuff, right?

The `system/build/os` value is resolved with call to [`cat /etc/*-release`](https://github.com/Oldes/Rebol3/blob/963d083a5e7fbf2c96ef9a60be386af7ff4402c0/make/tools/utils.reb#L173-L187) when being compiled (not the _bootstrap_ version, but the real one, using the proper builder script!).

I cannot help with Orx, as to be honest, I never even tried it and I have no idea how it is using Rebol and if its scripts are compatible with my Rebol branch.

**Comment by iarwain** [comment]
> I cannot help with Orx, as to be honest, I never even tried it and I have no idea how it is using Rebol and if its scripts are compatible with my Rebol branch.

Hopefully that is the case. =)
Thank you for your work on it, by the way, it's been a life savior for me, ever since I migrated orx's scripts from Ren-C about 2 years ago.

Going back to the issue at hand, I'm probably not using the right way to detect the runtime OS.
Maybe you could suggest a better approach, @Oldes? I simply need to know on what platform the script is running (Windows, MacOS or Linux), nothing more (something to replace checking the 4th digit of the `system/version` of the olden days).

**Comment by Oldes** [comment]
I think that @Ashvith10 is [using his _bootstrap_ version of Rebol](https://github.com/orx/orx/issues/86#issuecomment-1832338488), which is not correct.. the bootstrap version is using pre-generated sources, so it reports `system/build/os` to be `debian`, because that is the OS I used to generate the bootstrap. The _boostrap_ binary should be used just as a _stage-0_. Once compiled the _bootstrap_, it should be used to [compile the real version](https://github.com/Siskin-framework/Rebol/actions/runs/7035646575/workflow#L39-L43).

**Comment by Oldes** [comment]
@iarwain [this is not correct](https://github.com/orx/orx/blob/c99ca66ec358ce5bcb7d7106b234c5e1c12b2c88/code/build/rebol/setup.r#L32-L34). Use the `system/platform`.

**Comment by Oldes** [comment]
I am using [these helpers](https://github.com/Siskin-framework/Builder/blob/acc718db316621dae0bdd41c23e3cca53dd95a68/tree/rebol/Siskin/siskin.reb#L1971C1-L1979C108) in my Siskin Builder tool.

**Comment by iarwain** [comment]
> @iarwain [this is not correct](https://github.com/orx/orx/blob/c99ca66ec358ce5bcb7d7106b234c5e1c12b2c88/code/build/rebol/setup.r#L32-L34). Use the `system/platform`.

The interesting part is that `system/platform` was historically used: 

https://github.com/orx/orx/commit/fc15a72342c2f79a40b04e489a9a0041c215b320#diff-ca1c0171416b099db6bbbe71e25b69bb6394740886ec775f24ceef20fd0bde40L35

I made that change when migrating from Ren-C to your branch and I really don't remember why I did it.
If I have to venture a guess, it might have been because it looked more convenient (all lower case) and I did not realize until now that it was relative to the machine that built the interpreter.

I'll fix that tonight, thanks for the insight.
